app_dart/lib/src/service/github_checks_service.dart

```diff
@@ -114,6 +127,20 @@ class GithubChecksService {
     return true;
   }
 
+  /// Check if we should rerun a failed build.
+  ///
+  /// A presubmit retry happens if and only if
+  ///   1) the user is an auto roller account
+  ///   2) the same check run has never been retried before
+  bool shouldRerun(push_message.BuildPushMessage buildPushMessage) {
+    final push_message.Build build = buildPushMessage.build!;
+    if (!buildPushMessage.userData.containsKey('user_login') ||
```

Comment:
Please avoid as much as possible this type of logic as it introduces just another path of execution for a subset of users. Let's make the implementation generic and consistent for everybody.